### PR TITLE
fix(mme): log cleanup & alignment amf and ngap tasks

### PR DESCRIPTION
**Pull request by ganeshg87**
test cases:
   - initial registration,
     pdu session establishment,
     deregistration

Signed-off-by: ganeshg87 <ganesh.gedela@wavelabs.ai>

## Summary
fix(mme): log cleanup & alignment amf and ngap tasks

## Additional Information

- [ ] This change is backwards-breaking

### CODE REVIEW

**Comment by ganeshg87** [comment]
Please find sample ngap & amf logs,  

[ngap_debug.log](https://github.com/magma/magma/files/6955628/ngap_debug.log)
[ngap_info.log](https://github.com/magma/magma/files/6955629/ngap_info.log)
[amf_debug.log](https://github.com/magma/magma/files/6955630/amf_debug.log)
[amf_info_log.log](https://github.com/magma/magma/files/6955631/amf_info_log.log)

File: lte/gateway/c/core/oai/tasks/amf/Registration.cpp [hunk 1]
```diff
@@ -696,26 +686,24 @@ static int registration_accept_t3550_handler(
...
+      OAILOG_WARNING(
           LOG_AMF_APP,
           "T3550: timer has expired retransmitting registration accept\n");
       amf_send_registration_accept(amf_ctx);
     } else {
       /* Abort the registration procedure */
       OAILOG_ERROR(
           LOG_AMF_APP,
-          "T3550: Maximum retires:%d, for registration accept done hence Abort "
+          "T3550: Maximum retires:%d, for registration accept done hence "
```
Comment (ardzoht) [inline lte/gateway/c/core/oai/tasks/amf/Registration.cpp]:
```suggestion
          "T3550: Maximum retries:%d, for registration accept done hence "
```

File: lte/gateway/c/core/oai/tasks/amf/Registration.cpp [hunk 2]
```diff
@@ -772,11 +760,9 @@ int amf_proc_registration_complete(amf_context_t* amf_ctx) {
   } else {
     OAILOG_WARNING(
         LOG_NAS_AMF,
-        "UE Context not found for "
+        "ue context not found for "
```
Comment (ardzoht) [inline lte/gateway/c/core/oai/tasks/amf/Registration.cpp]:
```suggestion
        "UE context not found for "
```

File: lte/gateway/c/core/oai/tasks/amf/amf_app_main.cpp [hunk 3]
```diff
@@ -58,16 +58,19 @@ static int handle_message(zloop_t* loop, zsock_t* reader, void* arg) {
   amf_app_desc_t* amf_app_desc_p = get_amf_nas_state(false);
   imsi64_t imsi64                = itti_get_associated_imsi(received_message_p);
 
+  OAILOG_INFO(
```
Comment (ardzoht) [inline lte/gateway/c/core/oai/tasks/amf/amf_app_main.cpp:61]: should this be DEBUG level?

File: lte/gateway/c/core/oai/tasks/amf/amf_app_ue_context.cpp [hunk 4]
```diff
@@ -54,7 +55,7 @@ void notify_ngap_new_ue_amf_ngap_id_association(
 
   OAILOG_FUNC_IN(LOG_AMF_APP);
   if (ue_context_p == NULL) {
-    OAILOG_ERROR(LOG_AMF_APP, " NULL UE context!\n");
+    OAILOG_ERROR(LOG_AMF_APP, " null ue context!\n");
```
Comment (ardzoht) [inline lte/gateway/c/core/oai/tasks/amf/amf_app_ue_context.cpp]:
```suggestion
    OAILOG_ERROR(LOG_AMF_APP, " UE context is null\n");
```

File: lte/gateway/c/core/oai/tasks/amf/amf_security_mode_control.cpp [hunk 5]
```diff
@@ -305,11 +300,11 @@ int amf_proc_security_mode_control(
 
   OAILOG_DEBUG(
       LOG_NAS_AMF,
-      "Initiating security mode control procedure, "
-      "KSI = %d\n",
+      "initiating security mode control procedure, "
```
Comment (ardzoht) [inline lte/gateway/c/core/oai/tasks/amf/amf_security_mode_control.cpp]: for this and the other logs in general as well, why were these changed to start with lower case? in my opinion for cleanliness these all should start with capital letter, to follow the OAI codebase logging format

**Review by ardzoht — CHANGES_REQUESTED** [review]

File: lte/gateway/c/core/oai/tasks/amf/Registration.cpp (hunk 1 above)
Comment (ganeshg87) [reply lte/gateway/c/core/oai/tasks/amf/Registration.cpp]: addressed comment and resolved.

File: lte/gateway/c/core/oai/tasks/amf/Registration.cpp (hunk 2 above)
Comment (ganeshg87) [reply lte/gateway/c/core/oai/tasks/amf/Registration.cpp]: addressed comment, changed as you suggested.

File: lte/gateway/c/core/oai/tasks/amf/amf_app_main.cpp (hunk 3 above)
Comment (ganeshg87) [reply lte/gateway/c/core/oai/tasks/amf/amf_app_main.cpp:61]: AMF task received and sent out messages defined as INFO level.
I mean interaction messages between AMF--NGAP, AMF--SubscriberDB, AMF--SMF.

please have a look AMF_INFO log messages  :

image

Suggest me if the improvement is needed?

File: lte/gateway/c/core/oai/tasks/amf/amf_app_ue_context.cpp (hunk 4 above)
Comment (ganeshg87) [reply lte/gateway/c/core/oai/tasks/amf/amf_app_ue_context.cpp]: comment addressed.

File: lte/gateway/c/core/oai/tasks/amf/amf_security_mode_control.cpp (hunk 5 above)
Comment (ganeshg87) [reply lte/gateway/c/core/oai/tasks/amf/amf_security_mode_control.cpp]: comment addressed.

all log statements changed as suggested, starts with a capital letters.

**Comment by ganeshg87** [comment]
Please find the latest sample logs attached
[mme_11_Aug_2021.log](https://github.com/magma/magma/files/6966997/mme_11_Aug_2021.log)
[amf_debug.log](https://github.com/magma/magma/files/6966998/amf_debug.log)
[amf_info.log](https://github.com/magma/magma/files/6966996/amf_info.log)

File: lte/gateway/c/core/oai/tasks/amf/amf_app_main.cpp (hunk 3 above)
Comment (ardzoht) [reply lte/gateway/c/core/oai/tasks/amf/amf_app_main.cpp:61]: let's keep this as of now

**Review by ardzoht — APPROVED** [review]
lgtm

**Review by ulaskozat — APPROVED** [review]
LGTM
